src/code_80850_len_3060.c

```diff
@@ -268,7 +268,108 @@ void enforce_hpfp_limits(void) {
     }
 }
 
-INCLUDE_ASM(s32, "code_80850_len_3060", initialize_status_menu);
+void initialize_status_menu() {
+    UiStatus* uiStatus = &gUIStatus;
+    PlayerData* playerData = &gPlayerData;
+    s32 var0;
+    s32 var1;
+    s32 new_var;
+
+    uiStatus->drawPosX = 0x0C;
+    D_8010CD10 = 0;
+    D_8010CD12 = 0;
+    uiStatus->drawPosY = -0x64;
+    uiStatus->hidden = 0;
+    uiStatus->showTimer = 0xD2;
+    uiStatus->unk_3B[0] = 0;
+    uiStatus->ignoreChanges = 0;
+    uiStatus->unk_45[0] = 0;
+    uiStatus->unk_45[1] = 0;
+    uiStatus->hpBlinking = 0;
+    uiStatus->hpBlinkCounter = 0;
+    uiStatus->hpBlinkTimer = 0;
+    uiStatus->fpBlinking = 0;
+    uiStatus->fpBlinkCounter = 0;
+    uiStatus->fpBlinkTimer = 0;
+    uiStatus->spBlinking = 0;
+    uiStatus->spBlinkCounter = 0;
+    uiStatus->coinsBlinking = 0;
+    uiStatus->coinsBlinkCounter = 0;
+    uiStatus->coinsBlinkTimer = 0;
+    uiStatus->disabled = 0;
+    uiStatus->starpointsBlinking = 0;
+    uiStatus->displayHP = playerData->curHP;
+    uiStatus->displayFP = playerData->curFP;
+    uiStatus->displayCoins = playerData->coins;
+    uiStatus->displayStarpoints = playerData->starPoints;
+    uiStatus->displaySP = playerData->specialBarsFilled;
+    uiStatus->starpointsBlinkCounter = 0;
+    uiStatus->unk_6C[2] = -1;
+    uiStatus->unk_3B[1] = 0;
+    uiStatus->unk_57[0] = 0;
+    uiStatus->unk_57[1] = 0;
+    uiStatus->unk_57[2] = 0;
+    uiStatus->unk_57[3] = 0;
+    uiStatus->unk_6C[0] = 0;
+    uiStatus->unk_6C[1] = 0;
+    uiStatus->iconIndex12 = -1;
+    func_800E97B8();
+    var0 = create_icon(&D_80108248);
```

Comment:
I think each of these blocks that start with var0 and end with clear_icon_flags should be separated by an empty line, just to make it clear that they're separate little things